### PR TITLE
chore: upgrade deps following tokio 1.0

### PR DESCRIPTION
**Pull request by koivunej**
This was surprisingly easy, nothing special to call out.

I'll post a follow-up to this with all of the rustfmt/whitespace changes.

### CODE REVIEW

**Comment by koivunej** [comment]
Strange, once again this happened that local clippy reported nothing but the ci fails. Lets see which of these are issues for 1.45 and latest stable...

**Comment by koivunej** [comment]
Next up: beta clippy chores.

**Comment by koivunej** [comment]
So the `beta` and `nightly` clippy lints now trigger an error on stable because there is a similar lint. This can most likely be worked around, but also the streamlining of the CI around the MSRV (1.45.0) starts to sound tempting.

**Comment by koivunej** [comment]
Ok, trying another way around, I dont think there's a way to adjust the 1.45.0 clippys handling of future lint names.

**Comment by partim** [comment]
Changing `master::scan::Scanner::ok` to just `fn ok(&mut self)` and then changing the calls into `{ self.ok(); Ok(()) }` might be easier at this point. (If I didn’t miss anything, all the calls are for `Scanner::ok::<()>`).

**Comment by koivunej** [comment]
@parlin agreed as I am currently looking at what happens with `Ok(self.ok(()))` which triggers another "passing unit value to function" lint from clippy or rustc.

**Comment by partim** [comment]
This function certainly is, er, creative. Apologies for turning a seemingly straightforward task into a wild goose chase!

**Comment by koivunej** [comment]
Lets see if it works now... I'll squash that back and forth once we have a next issue.

**Comment by koivunej** [comment]
Perhaps with the NOOP TODO function, it would be better to have `todo!()`.

**Comment by koivunej** [comment]
Yeah, I think this is now ready for review. I am thinking all of the refactorings are quite clear, at least from the single-commit view. Nothing really perilous except for the `!(0x20...0x7f).contains(&ch)` which I was sure to copy paste (and in the other case change `b' '` into 0x20 for "uniform").

File: src/resolv/stub/conf.rs [hunk 1]
```diff
@@ -452,10 +452,16 @@ impl ResolvConf {
         words: SplitWhitespace
     ) -> Result<(), Error> {
         // XXX TODO
-        let _ = words; 
-        Ok(())
+        let _ = words;
+        let ret = Ok(());
+
+        if ret.is_err() {
```
Comment (partim) [inline src/resolv/stub/conf.rs:458]: Maybe it is better to comment out `parse_sortlist` entirely and replace its call in the match with a noop.

File: Changelog.md [hunk 2]
```diff
@@ -10,13 +10,15 @@ Breaking
...
   `domain::base::octets::OctetsBuilder`. It’s method `into_octets` is now
-  available as `freeze` on `OctetsBuilder. ([#75])
+  available as `freeze` on `OctetsBuilder`. ([#75])
 
 * Supprt for extended errors defined in [RFC 8914]. ([#79] by [@xofyarg])
 * New traits `domain::base::octets::OctetsFrom` and `OctetsInto` to
   convert types that are generic over octets sequences between different
   octets sequences. ([#77])
 
+* Upgrade to tokio 1.0, bytes 1.0, and latest of other dependencies ([#84])
```
Comment (partim) [inline Changelog.md:20]: You may or may not want to name-check yourself here. So far, I’ve always did that for ‘outside contributions.’

File: src/resolv/stub/conf.rs (hunk 1 above)
Comment (koivunej) [reply src/resolv/stub/conf.rs:458]: My preference would be to replace the match or the fn body with `todo!()` but I'll comment it out.

**Comment by partim** [comment]
I forgot to say this over all the Clippy kerfuffle: Thank you for the PR!

File: Changelog.md (hunk 2 above)
Comment (koivunej) [reply Changelog.md:20]: This sounds like a good way. I'll add myself here.

**Comment by koivunej** [comment]
> Thank you for the PR!

You're welcome. We are using `domain` in [`rust-ipfs`](https://github.com/rs-ipfs/rust-ipfs) in dnslink resolution; I didn't yet check if there are any other changes needed for us in the breaking but surely they can be handled.

**Comment by partim** [comment]
Cool! If there is anything you need or want or wish you had, let us know!

**Comment by koivunej** [comment]
@partim: I may have done a mistake by raising the smallvec version requirement needlessly, and while working on this, it seems that futures = "0.3.9" has been yanked replaced by "0.3.10". I'll see what would be a good way forward with these.

**Comment by partim** [comment]
Futures should be fine. Cargo will just resolve to 0.3.10, anyway.

I don’t think raising smallvec is an issue either. As far as semver goes, anyone requiring 1.2 should be fine with 1.6, too.

**Comment by koivunej** [comment]
> anyone requiring 1.2 should be fine with 1.6, too

Now that I double checked I may have misunderstood the conflict given to me by cargo while upgrading libp2p. Still, the #86 fixes the issue, while adding the additional minimal-version checks which should make the build more future proof, and need to think about this less in the future.

**Pull request merged**